genai-perf/tests/test_cli.py

```diff
@@ -133,27 +133,27 @@ def test_help_version_arguments_output_and_exit(
             ),
             (
                 ["--endpoint-type", "completions", "--service-kind", "openai"],
-                {"endpoint": "v1/completions"},
+                {"endpoint": None},
                 {"endpoint.custom": "v1/completions"},
             ),
             (
                 ["--endpoint-type", "chat", "--service-kind", "openai"],
-                {"endpoint": "v1/chat/completions"},
+                {"endpoint": None},
                 {"endpoint.custom": "v1/chat/completions"},
             ),
             (
                 ["--endpoint-type", "multimodal", "--service-kind", "openai"],
-                {"endpoint": "v1/chat/completions"},
+                {"endpoint": None},
                 {"endpoint.custom": "v1/chat/completions"},
             ),
             (
                 ["--endpoint-type", "rankings", "--service-kind", "openai"],
-                {"endpoint": "v1/ranking"},
+                {"endpoint": None},
                 {"endpoint.custom": "v1/ranking"},
             ),
             (
                 ["--endpoint-type", "image_retrieval", "--service-kind", "openai"],
-                {"endpoint": "v1/infer"},
+                {"endpoint": None},
```

Comment:
what is the difference between endpoint and endpoint.custom? is this something that user sets? why are endpoints here all None?

Reply:
It feels like this consolidation requires a unit testing overhaul. If we're moving a lot from the CLI to the config, we need to not just update the unit tests we have (like this one is mostly useless), but test that a config option passed in via the CLI results in the config having the right value later after the checking/infer happens.

We'd also want to make sure that both the success and failure unit tests are comprehensively tested, like we had for the CLI.

Reply:
Also, for `endpoint.custom`... is that verbiage correct? It sounds like that's not really a custom endpoint, it's just the endpoint.

Reply:
Everything in the CLI tests are still valid. It's just that the logic was moved into the Config class. When I added the Confiig class I updated test CLI to test both the argparse as well as the Config values. Any changes here (like the one here) reflect that the args value is no longer set (since the duplicated inference logic was removed), but shows that the inferencing is still being done (the config value is checked for correctness).

Reply:
Yes, `endpoint.custom` is correct.  If you look at parser.py and config_endpoint.py you'll note that the descriptions match (and I would argue that the name in config more closely represents the option's description).